### PR TITLE
PhyloBuddy: add in ascending and descending sorting of nodes

### PR DESCRIPTION
**Pull request by ctSkennerton**

### CODE REVIEW

**Comment by biologyguy** [comment]
Hi Connor,
Thanks for the pull request. Travis is failing, but it's because of a recent dependency issue that's cropped up which I haven't had time to address yet. I don't see anything in your code that would be causing an actual problem. I'm going to merge your contribution into the develop branch, however, instead of master, because you're changing the API. So 'sort' will be part of version 1.3. If I don't get to it tonight, I'll do it tomorrow.
If you're game, would you mind writing a couple unit tests and the wiki pages for your new functions?
-Steve

**Comment by ctSkennerton** [comment]
Sure I can write a couple of tests and add in some documentation

**Comment by biologyguy** [comment]
Also, what do you think about merging these two functions into a single function, and calling it 'ladderize'? It would be more in line with other functions like [SeqBuddy's Order IDs](https://github.com/biologyguy/BuddySuite/wiki/SB-Order-IDs).

**Comment by ctSkennerton** [comment]
no problem at all, I like the ladderized idea. I split them in two as there was a "wish list" section that had them that way.

**Comment by biologyguy** [comment]
Past Steve and current Steve are contradicting one another ;)
Let's go with a single function though; I think it will be cleaner.

**Comment by ctSkennerton** [comment]
Can you give me a couple of pointers to your test suite? The test files use a lot of hashed strings for comparison and parametrized values so I'm a little confused by what's going on. I guess I want to have a starting tree, apply the ladderization and then compare it to the expected tree.

**Comment by biologyguy** [comment]
Ya, sorry about that. It was a choice between tons of comparison files or hashes. I went hashes.
Have you already seen the unit test wiki page?
https://github.com/biologyguy/BuddySuite/wiki/Unit-tests

Maybe just merge the functions like we discussed, and I'll bang out a couple tests. If you're thinking of becoming an active contributor to the project, we can use this as an example case for me to show you how the test suite works.

**Comment by ctSkennerton** [comment]
Ah, I had not read that. I've combined the two functions to one with the latest commit but I don't imagine that I'll make too many changes in the future, so it is perhaps better to do the tests yourself.

File: buddysuite/buddy_resources.py [hunk 1]
```diff
@@ -1932,12 +1932,12 @@ def isfile_override(path):
             "unroot": {"flag": "ur",
                                "action": "store_true",
                                "help": "Remove any roots"},
-            "descending_order": {"flag": "do",
-                                 "action": "store_true",
-                                 "help": "sort the child nodes in order of the number of their children, in decending order"},
-            "ascending_order": {"flag": "ao",
-                                 "action": "store_true",
-                                 "help": "sort the child nodes in order of the number of their children, in ascending order"},
+            "ladderize": {"flag": "ld",
```
Comment (biologyguy) [inline buddysuite/buddy_resources.py]: This should actually live between "hash_ids" and "list_ids" to keep alpha-order. Apparently hash_ids and list_ids are themselves out of order, so swap them around too.

Comment (ctSkennerton) [inline buddysuite/buddy_resources.py]: I've updated the alpha order

**Comment by ctSkennerton** [comment]
Also I've added in some initial documentation to the wiki

**Comment by biologyguy** [comment]
Thanks again Connor,
Your contribution has been merged into develop branch.
When you get a moment, please add your information to [license.txt](https://github.com/biologyguy/BuddySuite/blob/develop/LICENSE.txt). As BuddySuite is primarily a product of the US government, it is released directly into Public Domain. You are under no obligation to license your own contributions as such, but do be mindful that choosing a restrictive license will mean I'll need to strip your code from the main repository before the next release to prevent any conflicts.
Best,
-Steve

**Pull request closed without merging**